File: backend/app/schemas/taxonomy.py
from pydantic import BaseModel


class TaxonomyNodeRecord(BaseModel):
    category_id: int
    category_name: str
    parent_id: int | None
    level: int
    path_ids: str
    path_names: str
    category_group_id: str | None = None
    category_pids: str | None = None
    category_group_name: str | None = None
    syn_list: str | None = None
    is_leaf: int = 0


class BuildTreeResult(BaseModel):
    file_id: int
    node_count: int
    root_count: int
    max_depth: int
    max_children_count: int
    leaf_count: int
    non_leaf_count: int
    missing_parent_count: int
    duplicate_name_count: int
    synonym_non_empty_count: int


class OverviewResult(BuildTreeResult):
    version_id: int

File: frontend/src/api/taxonomy.ts
import { apiGet } from './client'

export type TaxonomyNode = {
  id?: number
  version_id?: number
  category_id: number
  category_name: string
  parent_id?: number | null
  level?: number
  path_ids?: string
  path_names?: string
  syn_list?: string | null
  is_leaf?: number
  child_count?: number
  node_status?: string
}

export type TaxonomyOverview = {
  version_id: number
  file_id: number
  node_count: number
  root_count: number
  max_depth: number
  max_children_count: number
  leaf_count: number
  non_leaf_count: number
  missing_parent_count: number
  duplicate_name_count: number
  synonym_non_empty_count: number
}

export function getTaxonomyOverview(versionId: number) {
  return apiGet<TaxonomyOverview>(`/taxonomy/overview?version_id=${versionId}`)
}

export function getTreeLevel(versionId: number, parentId?: number) {
  const parent = parentId === undefined ? '' : `&parent_id=${parentId}`
  return apiGet<TaxonomyNode[]>(`/taxonomy/tree?version_id=${versionId}${parent}`)
}

export function searchTaxonomyNodes(versionId: number, query: string) {
  return apiGet<TaxonomyNode[]>(`/taxonomy/search?version_id=${versionId}&q=${encodeURIComponent(query)}`)
}
